cms/src/main/java/uk/nhs/digital/externalstorage/s3/S3ObjectDescriber.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.digital.externalstorage.s3;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import javax.xml.bind.DatatypeConverter;
+
+public class S3ObjectDescriber {
+
+    MessageDigest md = null;
+    String bucketName;
+
+    public S3ObjectDescriber(String bucketName) {
+        this.bucketName = bucketName;
+
+        try {
+            md = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ex) {
+            ex.printStackTrace();
```

Comment:
Unlikely as this exception is, it's better to re-throw it rather than snuffing it out. How about something like `throw new RuntimeException("Failed to initialise " + getClass().getSimpleName(), ex);`